qa/vagrant/src/main/java/org/elasticsearch/packaging/test/WindowsServiceTestCase.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.packaging.test;
+
+import junit.framework.TestCase;
+import org.elasticsearch.packaging.util.FileUtils;
+import org.elasticsearch.packaging.util.Platforms;
+import org.elasticsearch.packaging.util.ServerUtils;
+import org.elasticsearch.packaging.util.Shell;
+import org.elasticsearch.packaging.util.Shell.Result;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import static com.carrotsearch.randomizedtesting.RandomizedTest.assumeTrue;
+import static java.util.stream.Collectors.joining;
+import static org.elasticsearch.packaging.util.Archives.installArchive;
+import static org.elasticsearch.packaging.util.Archives.verifyArchiveInstallation;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public abstract class WindowsServiceTestCase extends PackagingTestCase {
+
+    private static final String DEFAULT_ID = "elasticsearch-service-x64";
+    private static final String DEFAULT_DISPLAY_NAME = "Elasticsearch " + FileUtils.getCurrentVersion() + " (elasticsearch-service-x64)";
+    private static String serviceScript;
+
+    private Shell sh;
+
+    @Before
+    public void createShell() {
+        sh = new Shell();
+    }
+
+    @BeforeClass
+    public static void ensureWindows() {
+        assumeTrue(Platforms.WINDOWS);
+    }
+
+    @After
+    public void uninstallService() {
+        sh.runIgnoreExitCode(serviceScript + " remove");
+    }
```

Comment:
I see why this needs to be in an `@After` method here. I think it would also make sense to do this in the routine that cleans everything up between tests https://github.com/elastic/elasticsearch/blob/2aba52de8f9315b0e384e1c657d7b0401d26a1b0/qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Cleanup.java#L90

Reply:
The call to the cleanup method (or in the cleanup method) is outside the windows service test case, which is the only thing that knows about the service existing.